src/ip_notifier.cpp

```diff
@@ -53,17 +55,21 @@ namespace libtorrent
 		TORRENT_UNUSED(ios);
 #elif defined TORRENT_WINDOWS
 		if (!m_hnd.is_open())
+#ifndef BOOST_NO_EXCEPTIONS
 			throw system_error(WSAGetLastError(), system_category());
+#else
+			std::terminate();
+#endif // BOOST_NO_EXCEPTIONS
 		m_ovl.hEvent = m_hnd.native_handle();
-#elif !TORRENT_USE_NETLINK
+#else
```

Comment:
This should not be changed. I don't want the TORRENT_UNUSED line on Linux, it is only for unsupported platforms.